bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/VerisureSession.java

```diff
@@ -0,0 +1,901 @@
+                return false;
+            }
+        }
+    }
+
+    public int sendCommand(String url, String data, @Nullable BigDecimal installationId) {
+        logger.debug("Sending command with URL {} and data {}", url, data);
+        configureInstallationInstance(installationId);
+        int httpResultCode = setSessionCookieAuthLogin();
+        if (httpResultCode == HttpStatus.OK_200) {
+            return postVerisureAPI(url, data);
+        } else {
+            return httpResultCode;
+        }
+    }
+
+    public boolean unregisterDeviceStatusListener(DeviceStatusListener deviceStatusListener) {
+        logger.debug("unregisterDeviceStatusListener for listener {}", deviceStatusListener);
+        return deviceStatusListeners.remove(deviceStatusListener);
+    }
+
+    public boolean registerDeviceStatusListener(DeviceStatusListener deviceStatusListener) {
+        logger.debug("registerDeviceStatusListener for listener {}", deviceStatusListener);
+        return deviceStatusListeners.add(deviceStatusListener);
+    }
+
+    public void dispose() {
+    }
+
+    public @Nullable VerisureThing getVerisureThing(String deviceId) {
+        return verisureThings.get(deviceId);
+    }
+
+    public HashMap<String, VerisureThing> getVerisureThings() {
+        return verisureThings;
+    }
+
+    public @Nullable String getCsrf() {
+        return csrf;
+    }
+
+    public @Nullable String getPinCode() {
+        return pinCode;
+    }
+
+    public String getApiServerInUse() {
+        return apiServerInUse;
+    }
+
+    public void setApiServerInUse(String apiServerInUse) {
+        this.apiServerInUse = apiServerInUse;
+    }
+
+    public String getNextApiServer() {
+        apiServerInUseIndex++;
+        if (apiServerInUseIndex > (APISERVERLIST.size() - 1)) {
+            apiServerInUseIndex = 0;
+        }
+        return APISERVERLIST.get(apiServerInUseIndex);
+    }
+
+    public void configureInstallationInstance(@Nullable BigDecimal installationId) {
+        logger.debug("Attempting to fetch CSRF and configure installation instance");
+        try {
+            csrf = getCsrfToken(installationId);
+            logger.debug("Got CSRF: {}", csrf);
+            // Set installation
+            String url = SET_INSTALLATION + installationId.toString();
+            logger.debug("Set installation URL: {}", url);
+            httpClient.GET(url);
+        } catch (ExecutionException e) {
+            logger.warn("Caught ExecutionException {}", e.getMessage(), e);
+        } catch (InterruptedException e) {
+            logger.warn("Caught InterruptedException {}", e.getMessage(), e);
+        } catch (TimeoutException e) {
+            logger.warn("Caught TimeoutException {}", e.getMessage(), e);
+        }
+    }
+
+    public @Nullable String getCsrfToken(@Nullable BigDecimal installationId) {
+        String subString = null;
+        String html = null;
+        String url = SETTINGS + installationId.toString();
```

Comment:
installationId is nullable but you are calling toString() on it?

Reply:
No it should not be nullable, I'll fix!